plugins/alembic/module/vtkF3DAlembicReader.cxx

```diff
@@ -20,51 +22,334 @@
 #pragma warning(pop)
 #endif
 
+using IndicesContainer = std::vector<int32_t>;
+using V3fContainer = std::vector<Alembic::Abc::V3f>;
+using AttributesContainer = std::map<std::string, V3fContainer>;
+using PerFaceWavefrontIndicesTripletsContainer = std::vector<Alembic::Abc::V3i>;
+using PerMeshWavefrontIndicesTripletsContainer =
+  std::vector<PerFaceWavefrontIndicesTripletsContainer>;
+
+const size_t P_indices_offset = 0;
+const size_t uv_indices_offset = 1;
+const size_t N_indices_offset = 2;
+
+struct IntermediateGeometry
+{
+  AttributesContainer _attributes;
+  PerMeshWavefrontIndicesTripletsContainer _indices;
+  bool _uv_is_facevarying{false};
+  bool _N_is_facevarying{false};
+};
+
 class vtkF3DAlembicReader::vtkInternals
 {
-public:
-  vtkSmartPointer<vtkPolyData> ProcessIPolyMesh(const Alembic::AbcGeom::IPolyMesh& pmesh)
+  void SetupIndicesStorage(const Alembic::AbcGeom::Int32ArraySamplePtr& face_vertex_counts,
+    PerMeshWavefrontIndicesTripletsContainer& extracted_indices)
   {
-    vtkNew<vtkPoints> points;
-    vtkNew<vtkCellArray> polys;
-    vtkNew<vtkPolyData> polydata;
 
-    Alembic::AbcGeom::IPolyMeshSchema::Sample samp;
-    if (pmesh.getSchema().getNumSamples() > 0)
+    for (size_t i = 0; i < face_vertex_counts->size(); i++)
     {
-      pmesh.getSchema().get(samp);
+      auto this_face_vertex_count = face_vertex_counts->get()[i];
+      PerFaceWavefrontIndicesTripletsContainer per_face_indices(this_face_vertex_count);
+      extracted_indices.push_back(per_face_indices);
+    }
+  }
 
-      Alembic::AbcGeom::P3fArraySamplePtr positions = samp.getPositions();
-      Alembic::AbcGeom::Int32ArraySamplePtr indices = samp.getFaceIndices();
-      Alembic::AbcGeom::Int32ArraySamplePtr counts = samp.getFaceCounts();
+  template<typename I>
+  void UpdateIndices(const I& attribute_indices, size_t indices_offset, bool do_reverse_rotate,
+    PerMeshWavefrontIndicesTripletsContainer& mesh_indices)
+  {
+    auto face_indices_counter = 0;
+    for (auto & per_face_indices : mesh_indices)
+    {
+      auto this_face_vertex_count = per_face_indices.size();
+      IndicesContainer this_face_indices;
+      // Perform the collection first
+      for (size_t j = 0; j < this_face_vertex_count; j++)
```

Comment:
Not really. Using a range-based loop will create a problem because the auto variable will be marked as unused.

Moreover, the `this_face_vertex_count` will be use further down round line 80